### PR TITLE
Contract Renewal

### PR DESCRIPTION
**Pull request by bookchin**
Implements a job that can be run at a chosen interval to scan the contracts collection and renew contracts that are still active.

* Depends https://github.com/Storj/complex/pull/47
* Adds a new `test-integration` directory for tests to run without DB stubs

### CODE REVIEW

**Comment by braydonf** [comment]
I think we can create and index with [multikey indexes](https://docs.mongodb.com/manual/core/index-multikey/) for `store_end` so that contracts that are expiring can be queried directly. 

e.g. with `ShardSchema.index({'contracts.contract.store_end': 1});`

**Comment by braydonf** [comment]
I've added the index here: https://github.com/Storj/service-storage-models/pull/77/files

File: bin/storj-contract-renew.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+ */
+function handleCursorData(shard) {
+  let needsRenewal = [];
+
+  cursor.pause();
+
+  for (let contractInfo of shard.contracts) {
+    let { contract: contractObj, nodeID } = contractInfo;
+    let contract = new storj.Contract(contractObj);
+    let willExpireNext24 = (contract.get('store_end') - HOURS_24) <= NOW;
```
Comment (braydonf) [inline bin/storj-contract-renew.js]: Is this necessary if we're querying by store_end?

Comment (bookchin) [inline bin/storj-contract-renew.js]: Nope, that's leftover from before the query/index change.

Comment (bookchin) [inline bin/storj-contract-renew.js]: Fixed

File: bin/storj-contract-renew.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    shard.save(next);
+  });
+}
+
+function renewContract([contact, contract], next) {
+  network.renewContract(contract, contact, (err) => {
+    if (err) {
+      counter.errored++;
```
Comment (braydonf) [inline bin/storj-contract-renew.js]: Would it be useful to log the reason why here?

**Comment by braydonf** [comment]
LGTM. I haven't run through and tested yet.

File: bin/storj-contract-renew.js (hunk 2 above)
Comment (bookchin) [inline bin/storj-contract-renew.js]: Updated to aggregate errors in a list with contact, contract, and error message to be included in the result output.

**Comment by braydonf** [comment]
Okay, ran the tests locally. Looks good. :)

**Pull request merged**